### PR TITLE
fix(base-trend-widget): change to setup script and convert data with helper

### PR DESCRIPTION
**Pull request by yuda110**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [x] Feature improvement
- [x] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### Description
스크린샷 2022-11-24 오전 10 26 41

- change from normal script to setup script
- create `getRefinedXYChartData` helper

### CODE REVIEW

File: src/services/dashboards/widgets/_base/base-trend/BaseTrendWidget.vue [hunk 1]
```diff
@@ -1,211 +1,178 @@
...
+                           :items="state.chartData"
+                           :currency="state.options.currency"
+                           :currency-rates="props.currencyRates"
+        />
+    </widget-frame>
 </template>
 
-<script lang="ts">
-import type { PropType } from 'vue';
+<script setup lang="ts">
```
Comment (yuda110) [inline src/services/dashboards/widgets/_base/base-trend/BaseTrendWidget.vue:29]: changed to **setup script**!

Comment (sulmoJ) [inline src/services/dashboards/widgets/_base/base-trend/BaseTrendWidget.vue:29]: 🙄

File: src/services/dashboards/widgets/_base/base-trend/BaseTrendWidget.vue [hunk 2]
```diff
@@ -1,211 +1,178 @@
...
+                />
+            </p-data-loader>
+        </div>
+
+        <widget-data-table :loading="state.loading"
+                           :fields="state.tableFields"
+                           :items="state.chartData"
+                           :currency="state.options.currency"
+                           :currency-rates="props.currencyRates"
+        />
```
Comment (sulmoJ) [inline src/services/dashboards/widgets/_base/base-trend/BaseTrendWidget.vue:25]: I'll try to finish it as soon as possible...!

**Review by sulmoJ — APPROVED** [review]
LGTM!!

**Pull request merged**